test/banned_commands.bats

```diff
@@ -4,8 +4,8 @@ load test_helpers
 
 banned_commands=(
     realpath
-    # readlink on OSX behaves differently from readlink on other Unix systems
-    readlink
+    # readlink -f on OSX behaves differently from readlink -f on other Unix systems
+    "readlink -f"
```

Comment:
This isn't going to catch all uses of `readlink -f`.

Reply:
Agree with @Stratus3D, it's better to leave `readlink` here.
@deiga, could you please revert this change.

Reply:
I'm not sure how to do that. The `normalize_readlink` function uses `readlink` and thus wouldn't pass the banned commands test.
Unless we build some exceptions logic to the banned commands test

Reply:
Could something like `readlink.+-.*f.+"` be good enough? And assume that some quoted variable reference follows the readlink invocation?

Reply:
@deiga To fix the test, you'll want to add some comment at the end of the line: `# asdf_allow: readlink`

That's probably the best way to go forward to absolutely ensure there are 0 chances of missing a `readlink -f` for the future

Reply:
Yeah I think a regex like you have there would work @hyperupcall.